src/net_processing.cpp

```diff
@@ -3557,12 +3617,15 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                   peer->m_starting_height, addrMe.ToStringAddrPort(), fRelay, pfrom.GetId(),
                   remoteAddr, (mapped_as ? strprintf(", mapped_as=%d", mapped_as) : ""));
 
-        int64_t nTimeOffset = nTime - GetTime();
-        pfrom.nTimeOffset = nTimeOffset;
+        peer->m_time_offset = nTime - GetTime();
         if (!pfrom.IsInboundConn()) {
-            // Don't use timedata samples from inbound peers to make it
-            // harder for others to tamper with our adjusted time.
-            AddTimeData(pfrom.addr, nTimeOffset);
+            m_outbound_time_offsets.Add(peer->m_time_offset);
+
+            auto median = m_outbound_time_offsets.Median();
+            if (median > m_opts.time_offset_warn_threshold.count() ||
+                median < -m_opts.time_offset_warn_threshold.count()) {
+                SetMedianTimeOffsetWarning();
+            }
```

Comment:
This does seem to change warning behaviour, in that:
1. it isn't logged anymore
2. there's no more pop-up message for bitcoin-qt

Is that intentional?